<!doctype html><html lang=en><meta charset=utf-8><meta name=viewport content="width=device-width"><title>Reference genome | Ziwei Pan 潘紫薇</title><link rel=stylesheet href=https://ziweipan.me/css/eureka.min.css><script defer src=https://ziweipan.me/js/eureka.min.js></script><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=preload href="https://fonts.googleapis.com/css2?family=Lora:wght@400;600;700&family=Noto+Serif+SC:wght@400;600;700&display=swap" as=style onload="this.onload=null;this.rel='stylesheet'"><link rel=stylesheet href=https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@10.1.0/build/styles/solarized-light.min.css media=print onload="this.media='all';this.onload=null" crossorigin><script defer src=https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@10.1.0/build/highlight.min.js crossorigin></script><script defer src=https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.14.0/js/all.min.js integrity="sha256-uNYoXefWRqv+PsIF/OflNmwtKM4lStn9yrz2gVl6ymo=" crossorigin></script><link rel=stylesheet href=https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.css integrity=sha384-AfEj0r4/OFrOo5t7NnNe46zW/tFgW6x/bCJG8FqQCEo3+Aro6EYUG4+cU+KJWu/X media=print onload="this.media='all';this.onload=null" crossorigin><script defer src=https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.js integrity=sha384-g7c+Jr9ZivxKLnZTDUhnkOnsh30B4H0rpLUpJ4jAIKs4fnJI+sEnkvrMWph2EDg4 crossorigin></script><script defer src=https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/contrib/auto-render.min.js integrity=sha384-mll67QQFJfxn0IYznZYonOWZ644AWYC+Pt2cHqMaRhXVrursRwvLnLaebdGIlYNa crossorigin></script><script>document.addEventListener("DOMContentLoaded",function(){renderMathInElement(document.body,{delimiters:[{left:"$$",right:"$$",display:true},{left:"$",right:"$",display:false},{left:"\\(",right:"\\)",display:false},{left:"\\[",right:"\\]",display:true}],});});</script><link rel=icon type=image/png sizes=32x32 href=https://ziweipan.me/images/icon_hub90c5072bafd0120d2324ea120261cc6_12329_32x32_fill_box_center_2.png><link rel=apple-touch-icon sizes=180x180 href=https://ziweipan.me/images/icon_hub90c5072bafd0120d2324ea120261cc6_12329_180x180_fill_box_center_2.png><meta name=description content="A brief introduction for reference genome in FASTA and other format."><script type=application/ld+json>{"@context":"https://schema.org","@type":"BreadcrumbList","itemListElement":[{"@type":"ListItem","position":1,"name":"Posts","item":"https://ziweipan.me/en/posts/"},{"@type":"ListItem","position":2,"name":"Reference genome","item":"https://ziweipan.me/en/posts/reference_genome/"}]}</script><script type=application/ld+json>{"@context":"https://schema.org","@type":"Article","mainEntityOfPage":{"@type":"WebPage","@id":"https://ziweipan.me/en/posts/reference_genome/"},"headline":"Reference genome | Ziwei Pan 潘紫薇","datePublished":"2021-01-28T16:17:15-05:00","dateModified":"2021-02-24T22:27:28-05:00","wordCount":554,"publisher":{"@type":"Person","name":"Ziwei","logo":{"@type":"ImageObject","url":"https://ziweipan.me/images/avatar.png"}},"description":"\u003cp\u003eA brief introduction for reference genome in FASTA and other format.\u003c\/p\u003e"}</script><meta property="og:title" content="Reference genome | Ziwei Pan 潘紫薇"><meta property="og:type" content="article"><meta property="og:image" content="https://ziweipan.me/images/icon.png"><meta property="og:url" content="https://ziweipan.me/en/posts/reference_genome/"><meta property="og:description" content="A brief introduction for reference genome in FASTA and other format."><meta property="og:locale" content="en"><meta property="og:site_name" content="Ziwei Pan 潘紫薇"><meta property="article:published_time" content="2021-01-28T16:17:15-05:00"><meta property="article:modified_time" content="2021-02-24T22:27:28-05:00"><meta property="article:section" content="posts"><meta property="article:tag" content><body class="flex flex-col min-h-screen"><header class="fixed flex items-center w-full min-h-16 pl-scrollbar z-50 bg-secondary-bg shadow-sm"><div class="w-full max-w-screen-xl mx-auto"><script>let storageColorScheme=localStorage.getItem("lightDarkMode")
if(((storageColorScheme=='Auto'||storageColorScheme==null)&&window.matchMedia("(prefers-color-scheme: dark)").matches)||storageColorScheme=="Dark"){document.getElementsByTagName('html')[0].classList.add('dark')}</script><nav class="flex items-center justify-between flex-wrap px-4 py-4 md:py-0"><a href=/en/ class="mr-6 text-primary-text text-xl font-bold">Ziwei Pan 潘紫薇</a>
<button id=navbar-btn class="md:hidden flex items-center px-3 py-2" aria-label="Open Navbar">
<i class="fas fa-bars"></i></button><div id=target class="hidden block md:flex md:flex-grow md:justify-between md:items-center w-full md:w-auto text-primary-text z-20"><div class="md:flex md:h-16 text-sm md:flex-grow pb-4 md:pb-0 border-b md:border-b-0"><a href=/en/about/ class="block mt-4 md:inline-block md:mt-0 md:h-(16-4px) md:leading-(16-4px) box-border md:border-t-2 md:border-b-2 border-transparent mr-4">About</a>
<a href=/en/posts/ class="block mt-4 md:inline-block md:mt-0 md:h-(16-4px) md:leading-(16-4px) box-border md:border-t-2 md:border-b-2 selected-menu-item mr-4">Posts</a></div><div class=flex><div class="relative pt-4 md:pt-0"><div class="cursor-pointer hover:text-eureka" id=lightDarkMode><i class="fas fa-adjust"></i></div><div class="fixed hidden inset-0 opacity-0 h-full w-full cursor-default z-30" id=is-open></div><div class="absolute flex flex-col left-0 md:left-auto right-auto md:right-0 hidden bg-secondary-bg w-48 rounded py-2 border border-tertiary-bg cursor-pointer z-40" id=lightDarkOptions><span class="px-4 py-1 hover:text-eureka">Light</span>
<span class="px-4 py-1 hover:text-eureka">Dark</span>
<span class="px-4 py-1 hover:text-eureka">Auto</span></div></div></div></div><div class="fixed hidden inset-0 opacity-0 h-full w-full cursor-default z-0" id=is-open-mobile></div></nav><script>let element=document.getElementById('lightDarkMode')
if(storageColorScheme==null||storageColorScheme=='Auto'){document.addEventListener('DOMContentLoaded',()=>{switchMode('Auto')})}else if(storageColorScheme=="Light"){element.firstElementChild.classList.remove('fa-adjust')
element.firstElementChild.setAttribute("data-icon",'sun')
element.firstElementChild.classList.add('fa-sun')}else if(storageColorScheme=="Dark"){element.firstElementChild.classList.remove('fa-adjust')
element.firstElementChild.setAttribute("data-icon",'moon')
element.firstElementChild.classList.add('fa-moon')}
document.addEventListener('DOMContentLoaded',()=>{getcolorscheme();switchBurger();});</script></div></header><main class="flex-grow pt-16"><div class=pl-scrollbar><div class="w-full max-w-screen-xl lg:px-4 xl:px-8 mx-auto"><div class="grid grid-cols-2 lg:grid-cols-8 gap-4 lg:pt-12"><div class="col-span-2 lg:col-span-6 bg-secondary-bg rounded px-6 py-8"><h1 class="font-bold text-3xl text-primary-text">Reference genome</h1><div class="flex flex-wrap flex-row items-center mt-2 text-tertiary-text"><div class="mr-6 my-2"><i class="fas fa-calendar mr-1"></i><span>January 28, 2021</span></div><div class="mr-6 my-2"><i class="fas fa-clock mr-1"></i><span>2 min read</span></div><div class="mr-6 my-2"><i class="fas fa-folder mr-1"></i><a href=https://ziweipan.me/en/categories/bioinfo/ class=hover:text-eureka>bioinfo</a></div></div><div class=content><p>A brief introduction for reference genome in FASTA and other format.</p><h2 id=human-reference-genome>Human reference genome</h2><p><a href=https://www.ncbi.nlm.nih.gov/datasets/genomes/>Genomes – NCBI Datasets BETA</a>: Download a genome dataset including genome, transcript and protein sequence, annotation and a data report</p><p><strong>GRCh38</strong> (latest version -GRCh38.p13):</p><p>Introduction: <a href=https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.39>https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.39</a></p><p><a href=ftp://ftp.ensembl.org/pub/current_fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa.gz>ftp://ftp.ensembl.org/pub/current_fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa.gz</a></p><p>By chromosome: <a href=ftp://ftp.ensembl.org/pub/current_fasta/homo_sapiens/dna/>ftp://ftp.ensembl.org/pub/current_fasta/homo_sapiens/dna/</a></p><p><strong>GRCh37</strong> (latest version -GRCh37.p13):</p><p>Introduction: <a href=https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.13/>https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.13/</a></p><p><a href=ftp://ftp.ensembl.org/pub/grch37/current/fasta/homo_sapiens/dna/Homo_sapiens.GRCh37.dna_sm.primary_assembly.fa.gz>ftp://ftp.ensembl.org/pub/grch37/current/fasta/homo_sapiens/dna/Homo_sapiens.GRCh37.dna_sm.primary_assembly.fa.gz</a></p><p><a href=ftp://hgdownload.cse.ucsc.edu/goldenPath/hg19/hg19Patch13/hg19Patch13.fa.gz>ftp://hgdownload.cse.ucsc.edu/goldenPath/hg19/hg19Patch13/hg19Patch13.fa.gz</a></p><h2 id=download>Download</h2><h3 id=download-command>Download command</h3><p>! Attention: Remember to create the fasta index file for later usage.</p><p><strong>GRch38</strong>:</p><pre><code class=language-shell>wget ftp://ftp.ensembl.org/pub/current_fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa.gz \
-O -|gunzip -c &gt; $reference_dir/GRch38.fa
### Create the fasta index file for later usage
samtools faidx $reference_dir/GRch38.fa

&gt;chr1
NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN
NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN
</code></pre><p><strong>GRch37</strong>:</p><pre><code class=language-shell># Ensembl version:
wget ftp://ftp.ensembl.org/pub/grch37/current/fasta/homo_sapiens/dna/Homo_sapiens.GRCh37.dna.primary_assembly.fa.gz \
-O -|gunzip -c &gt; $reference_dir/GRCh37.fa
samtools faidx $reference_dir/GRCh37.fa

&gt;1 dna:chromosome chromosome:GRCh37:1:1:249250621:1 REF
NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN
NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN

or
# NCBI version:
wget ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_mammalian/Homo_sapiens/all_assembly_versions/GCF_000001405.25_GRCh37.p13/GCF_000001405.25_GRCh37.p13_genomic.fna.gz \
-O -|gunzip -c &gt; $reference_dir/GRCh37.new.fa
samtools faidx $reference_dir/GRCh37.new.fa

&gt;NC_000001.10 Homo sapiens chromosome 1, GRCh37.p13 Primary Assembly
NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN
NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN
</code></pre><h2 id=other-issue>Other issue</h2><h3 id=toplevel-fasta-vs-primary_assembly-fasta>toplevel fasta VS primary_assembly fasta</h3><ul><li>toplevel fasta</li></ul><p>Ensembl advise using the full toplevel fasta reference when doing CrossMapping, and for GRCh38, that file is HUGE, (36G instead of the usual 3G, containing all the ALT sequences). As CrossMap loads the reference into memory, this makes a big difference. However, the resulting mapped files are identical for mappings made with the toplevel file as with the primary_assembly file (which does not contain all the ALT seqs), so it is preferable to use the latter from a resources point of view.</p><ul><li>Primary assembly</li></ul><p>contains all toplevel sequence regions excluding haplotypes and patches. This file is best used for performing sequence similarity searches where patch and haplotype sequences would confuse analysis.</p><h3 id=different-fasta-format>Different FASTA format</h3><p>.fasta, .fas, .fa, .seq, .fsa: Generic FASTA</p><p>.fna: FASTA nucleic acid</p><p>.faa: FASTA amino acids, for protein sequences</p><p>.ffn: FASTA nucleotide coding regions for a genome</p><p>.mpfa: FASTA amino acides in multiple proteins</p><p>.frn: FASTA non-coding RNA</p><p>.fastq: FASTQ</p><h3 id=grch37-issue>GRCh37 issue</h3><ul><li>GRCh37 hg19 b37 humanG1Kv37 - Human Reference Discrepancies**</li></ul><p>There are 4 common &ldquo;hg19&rdquo; references, and they are NOT directly interchangeable:</p><p>-hg19 (<code>ucsc.hg19.fasta</code>, MD5sum: <code>a244d8a32473650b25c6e8e1654387d6</code>). hg19 is UCSC&rsquo;s variant of the official GRCh37 assembly.</p><p>-b37 (<code>Homo_sapiens_assembly19.fasta</code>, MD5sum: <code>886ba1559393f75872c1cf459eb57f2d)</code></p><p>-GRCh37 (<code>GRCh37.p13.genome.fasta</code>, MD5sum: <code>c140882eb2ea89bc2edfe934d51b66cc</code>)</p><p>-humanG1Kv37 (<code>human_g1k_v37.fasta</code>, MD5sum: <code>0ce84c872fc0072a885926823dcd0338</code>)</p><p>More details: <a href=https://gatk.broadinstitute.org/hc/en-us/articles/360035890711-GRCh37-hg19-b37-humanG1Kv37-Human-Reference-Discrepancies>https://gatk.broadinstitute.org/hc/en-us/articles/360035890711-GRCh37-hg19-b37-humanG1Kv37-Human-Reference-Discrepancies</a></p><ul><li>prefix &ldquo;chr&rdquo; issue</li></ul><p>hg19 or early releases of GRCh37 like <code>GRCh37-lite</code>, did not use chr-prefixes, but newer releases like <code>GRCh37.p13</code> adopted the chr-prefix, and use a newer mitochondrial (MT) sequence than hg19 does. chrM in hg19 is named MT in GRCh37.</p><pre><code>(base) [c-panz@sumner-log2 reference]$ head GRCh37.fa
&gt;1 dna_sm:chromosome chromosome:GRCh37:1:1:249250621:1 REF
nnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnn
nnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnn
nnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnn
</code></pre><p>!Attention: DO NOT use <code>sed</code>/<code>cat</code> or other find/replace tools to fix this problem yourself (unless you are extremely bandwidth-constrained or no option to download a properly-formatted reference FASTA).</p><p><strong>Modify chromosome names: with &ldquo;chr&rdquo; prefix or without</strong></p><p>Be very careful to use the command below:</p><pre><code># add chr
sed 's/^&gt;/&gt;chr/' GRCh37.fa &gt; GRCh37_new.fa

# or
awk 'BEGIN {OFS = &quot;\t&quot;} {$1=&quot;chr&quot;$1; print}'

# or A faster option, from the BBMap package:
bbrename.sh in=file.fasta out=renamed.fasta prefix=ch addprefix=t


# remove chr
sed 's/^chr//' GRCh37.fa &gt; GRCh37_new.fa
</code></pre><p>More discussion: <a href=https://www.biostars.org/p/119295/#119308>Question: Human dna reference file with no prefix &lsquo;chr&rsquo;</a></p><h2 id=reference>Reference</h2><p><a href=https://gatk.broadinstitute.org/hc/en-us/articles/360035891071>GATK: Reference genome</a></p><p><a href=https://gatk.broadinstitute.org/hc/en-us/articles/360035890951-Human-genome-reference-builds-GRCh38-or-hg38-b37-hg19>GATK: Human genome reference builds - GRCh38 or hg38 - b37 - hg19</a></p><p><a href=https://lh3.github.io/2017/11/13/which-human-reference-genome-to-use>Heng Li: Which human reference genome to use?</a></p><p><a href=https://bitesizebio.com/38335/get-to-know-your-reference-genome-grch37-vs-grch38/>Get to Know Your Reference Genome (GRCh37 vs GRCh38)</a></p><p><a href=https://wenlongshen.github.io/2020/03/26/Reference-Genome/>关于人类基因组的一些说明</a></p><p><a href=https://wabi-wiki.scilifelab.se/display/SHGATG/gatk+bundle+in+hg38>https://wabi-wiki.scilifelab.se/display/SHGATG/gatk+bundle+in+hg38</a></p><p><a href=https://github.com/jmzeng1314/bioinformatics123/blob/master/appendix/database.md>生物信息学数据库资源</a></p><p><a href=ftp://ftp.ensembl.org/pub/grch37/current/fasta/homo_sapiens/dna/README>ftp://ftp.ensembl.org/pub/grch37/current/fasta/homo_sapiens/dna/README</a></p></div><div class="flex flex-col md:flex-row md:justify-between -mx-2 mt-4 px-2 pt-4 border-t"><div></div><div class="md:text-right mt-4 md:mt-0"><span class="block font-bold">Next</span>
<a href=https://ziweipan.me/en/posts/jupyterlab_keyboard_shortcut/ class=block>JupyterLab Keyboard Shortcut</a></div></div></div><div class=col-span-2><div class="sticky top-16 z-10 hidden lg:block px-6 py-4 bg-primary-bg"><span class="text-lg font-semibold">On This Page</span></div><div class="sticky-toc hidden lg:block px-6 pb-6"><nav id=TableOfContents><ul><li><a href=#human-reference-genome>Human reference genome</a></li><li><a href=#download>Download</a><ul><li><a href=#download-command>Download command</a></li></ul></li><li><a href=#other-issue>Other issue</a><ul><li><a href=#toplevel-fasta-vs-primary_assembly-fasta>toplevel fasta VS primary_assembly fasta</a></li><li><a href=#different-fasta-format>Different FASTA format</a></li><li><a href=#grch37-issue>GRCh37 issue</a></li></ul></li><li><a href=#reference>Reference</a></li></ul></nav></div><script>window.addEventListener('DOMContentLoaded',()=>{enableStickyToc();});</script></div></div><script>document.addEventListener('DOMContentLoaded',()=>{hljs.initHighlightingOnLoad();})</script></div></div></main><footer class=pl-scrollbar><div class="w-full max-w-screen-xl mx-auto"><div class="text-center p-6 pin-b"><p class="text-sm text-tertiary-text">&copy; 2019-2021 Ziwei Pan &#183; Powered by the <a href=https://github.com/wangchucheng/hugo-eureka class=hover:text-eureka>Eureka</a> theme for <a href=https://gohugo.io class=hover:text-eureka>Hugo</a></p></div></div></footer></body></html>